vertical-pod-autoscaler/recommender/base/circularbuffer.go

```diff
@@ -0,0 +1,62 @@
+package base
+
+// A queue with a fixed maximum capacity. Elements are inserted/removed in the
+// FIFO order. Elements are removed from the buffer when it runs out of capacity
+// and a new element is inserted.
+type CircularBuffer interface {
```

Comment:
I create a minimal interface necessary to implement the operations we will use in recommender (see above). We can certainly extend it if needed.
Note: I also changed Head() to return nil if the buffer is empty.